### PR TITLE
Added OpenID Login

### PR DESCRIPTION
**Pull request by Revadike**
Usage:
```
const steamcommunity = require("steamcommunity");
const community = new steamcommunity();
community.login({
    "accountName": "username",
    "password": "password"
}, function(err) {
    console.log(err || "logged in");
    community.openidLogin("https://royalgamer06.ga/barter/?login", function(err, response, data) {
        if (err) return console.log("error:", err);
        console.log(data); // Logged into third party site
    });
});```

### CODE REVIEW

**Comment by Royalgamer06** [comment]
Perhaps out of scope of this module, but personally, I find it a very useful thing, if I need "Sign in through Steam" on a third party website.

**Comment by DoctorMcKay** [comment]
I do think this is out of scope. This might be better as its own npm module.

**Comment by Royalgamer06** [comment]
It does fit very nicely in the current module with not much code and coded in your style :)

**Comment by scholtzm** [comment]
I'd say it's only very slightly out of scope but it's a pretty sought out feature from the questions I have seen at /r/SteamBot.

@Royalgamer06 

`console.log(data); // Logged into third party site`

What's inside `data`?

**Comment by Royalgamer06** [comment]
The response HTML body.

**Comment by scholtzm** [comment]
Oops I totally missed the function signature just above.

This is pretty useful since you can just take the cookies from response object and reuse them on subsequent requests.

File: index.js [hunk 1]
```diff
@@ -249,6 +250,48 @@ SteamCommunity.prototype.oAuthLogin = function(steamguard, token, callback) {
 	}, "steamcommunity");
 };
 
+SteamCommunity.prototype.openidLogin = function(url, callback) {
+    var self = this;
+    self.loggedIn(function(err, loggedIn) {
```
Comment (scholtzm) [inline index.js]: Perhaps this check could be removed. Let the user handle this on their own.

Comment (scholtzm) [reply index.js]: For example, the snippet in the first comment makes an unnecessary request since we know for sure we are logged in.

Comment (Royalgamer06) [reply index.js]: You got a point. I just wanted it to be robust, but performance is a good quality too.

**Comment by Royalgamer06** [comment]
Hey, what if I argue that the OpenID login is a feature of the Steam Community and therefore fits with this project's scope.

**Comment by mert574** [comment]
I think it is slightly out of scope but it is actually helpful. Having it will not produce much to worry about

**Comment by Kladdkaka** [comment]
I think it is out of scope, but it is as many other people said, a very useful feature, I would suggest adding it. :)

**Comment by Fxlr8** [comment]
Merge this feature, please

**Comment by Revadike** [comment]
My extensive track record of closed PR's grows again. Sad.

**Comment by Revadike** [comment]
This is undoubtably a very useful function node-steamcommunity to provide. So, I ask you @DoctorMcKay you perhaps reconsider this PR. Maybe add it in v4.

**Pull request closed without merging**